packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -71,30 +72,48 @@ export class GutenbergEditorPage {
 	/**
 	 * Initialization steps to ensure the page is fully loaded.
 	 *
+	 * This method will perform the following actions:
+	 * 	- retrieves the editor frame, retrying up to 3 times as required
+	 * 	  with a browser page refresh.
+	 * 	- dismiss the welcome tour, if present.
+	 *
+	 * Calling this one method will typically be sufficient to ensure the
+	 * editor is prepared and ready for subsequent actions.
+	 *
 	 * @returns {Promise<Frame>} iframe holding the editor.
 	 */
 	async waitUntilLoaded(): Promise< Frame > {
-		await this.page.waitForLoadState( 'load' );
+		/**
+		 * Closure that is nearly identical to the `getEditorFrame` method but
+		 * one that operates on the Page object passed in as parameter instead
+		 * of the class property.
+		 *
+		 * This closure will first use Locator API to locate the editor frame,
+		 * then retrieve the ElementHandle from the locator.
+		 */
+		async function waitForEditorFrame( page: Page ): Promise< void > {
+			const locator = page.locator( selectors.editorFrame );
+			await locator.elementHandle( { timeout: iframeTimeout } );
+		}
 
-		const frame = await this.getEditorFrame();
-		// Traditionally we try to avoid waits not related to the current flow. However, we need a stable way to identify loading being done.
-		// NetworkIdle takes too long here, so the most reliable alternative is the title being visible.
-		await frame.waitForSelector( selectors.editorTitle );
+		await reloadAndRetry( this.page, waitForEditorFrame );
 
-		await this.dismissWelcomeTourIfPresent();
-		return frame;
+		return await this.dismissWelcomeTour();
 	}
 
 	/**
-	 * Dismisses the Welcome Tour (card) if it is present.
+	 * Dismisses the Welcome Tour (card) if it is present, then return the editor frame.
+	 *
+	 * @returns {Promise<Frame>} iframe holding the editor.
 	 */
-	async dismissWelcomeTourIfPresent(): Promise< void > {
+	async dismissWelcomeTour(): Promise< Frame > {
 		const frame = await this.getEditorFrame();
 		try {
-			await frame.click( selectors.welcomeTourCloseButton, { timeout: 5 * 1000 } );
+			await frame.click( selectors.welcomeTourCloseButton, { timeout: 3 * 1000 } );
```

Comment:
It would be great to know when this Welcome tour does come up as it's a constant 3 seconds added to each call now. Can it be verified via some `nux` request maybe? Then I think we could even use the [`page.on('request')`](https://playwright.dev/docs/api/class-page#page-event-request) event listener.

Reply:
I think you're onto something about the `nux` event. I located this request when loading the editor:

`https://public-api.wordpress.com/wpcom/v2/sites/198111396/block-editor/nux?_envelope=1`

which has the response:

```
{"body":{"show_welcome_guide":false,"variant":"tour"},"status":200,"headers":{"Allow":"GET, POST, PUT, PATCH"}}
```

I think inspecting this might be a good call.